prometheus-client-derive/src/registrant/attribute.rs

```diff
@@ -0,0 +1,135 @@
+        self
+    }
+
+    pub(super) fn merge(self, other: Self) -> syn::Result<Self> {
+        let mut merged = self;
+
+        if let Some(doc) = other.help {
+            // trim leading and trailing whitespace
+            // and add a space between the two doc strings
+            let mut acc = merged
+                .help
+                .unwrap_or_else(|| syn::LitStr::new("", doc.span()))
+                .value()
+                .trim()
+                .to_string();
+            acc.push(' ');
+            acc.push_str(doc.value().trim());
+            merged.help = Some(syn::LitStr::new(&acc, Span::call_site()));
+        }
+        if let Some(unit) = other.unit {
+            if merged.unit.is_some() {
+                return Err(syn::Error::new_spanned(
+                    merged.unit,
+                    "Duplicate `unit` attribute",
+                ));
+            }
+
+            merged.unit = Some(unit);
+        }
+        if let Some(rename) = other.rename {
+            if merged.rename.is_some() {
+                return Err(syn::Error::new_spanned(
+                    merged.rename,
+                    "Duplicate `rename` attribute",
+                ));
+            }
+
+            merged.rename = Some(rename);
+        }
+        if other.skip {
+            merged.skip = merged.skip || other.skip;
+        }
+
+        Ok(merged)
+    }
+}
+
+impl syn::parse::Parse for Attribute {
+    fn parse(input: syn::parse::ParseStream) -> syn::Result<Self> {
+        let meta = input.parse::<syn::Meta>()?;
+        let span = meta.span();
+
+        match meta {
+            syn::Meta::NameValue(meta) if meta.path.is_ident("doc") => {
+                if let syn::Expr::Lit(lit) = meta.value {
+                    let lit_str = syn::parse2::<syn::LitStr>(lit.lit.to_token_stream())?;
+                    Ok(Attribute::default().with_help(lit_str))
+                } else {
+                    Err(syn::Error::new_spanned(
+                        meta.value,
+                        "Expected a string literal for doc attribute",
+                    ))
+                }
+            }
+            syn::Meta::List(meta) if meta.path.is_ident("registrant") => {
+                let mut attr = Attribute::default();
+                meta.parse_nested_meta(|meta| {
+                    if meta.path.is_ident("unit") {
+                        let unit = meta.value()?.parse::<syn::LitStr>()?;
+
+                        if attr.unit.is_some() {
+                            return Err(syn::Error::new(
+                                meta.path.span(),
+                                "Duplicate `unit` attribute",
+                            ));
+                        }
+
+                        // unit should be lowercase
+                        let unit = syn::LitStr::new(
+                            unit.value().as_str().to_ascii_lowercase().as_str(),
+                            unit.span(),
+                        );
```

Comment:
There are two options:

1. Covert to lowercase silently.
2. Emit compilation error.

What do you think?